rasa/graph_components/validators/finetuning_validator.py

```diff
@@ -0,0 +1,341 @@
+            importer: a training data importer
+            domain: the domain
+            config: the configuration (including the recipe)
+            nlu: set to `False` if NLU part should not be validated
+            core: set to `False` if Core part should not be validated
+        Raises:
+            `InvalidConfigException` if there is a conflict
+        """
+        finetuning = self._execution_context.is_finetuning
+
+        if finetuning and not self._fingerprints:
+            raise InvalidConfigException(
+                f"Finetuning is enabled but the {self.__class__.__name__} "
+                f"does not remember seeing a training run. Ensure that you have "
+                f"trained your model at least once (with finetuning disabled) "
+                f"and ensure that the  {self.__class__.__name__} is part of the "
+                f"training graph. "
+            )
+
+        # check the rasa version
+
+        if finetuning:
+            old_rasa_version = self._fingerprints[FINGERPRINT_VERSION]
+            if version.parse(old_rasa_version) < version.parse(
+                MINIMUM_COMPATIBLE_VERSION
+            ):
+                raise InvalidConfigException(
+                    f"The minimum compatible Rasa Version is "
+                    f"{MINIMUM_COMPATIBLE_VERSION} but the model we attempt to "
+                    f"finetune has been generated with an older version "
+                    f"({old_rasa_version}."
+                )
+        self._fingerprints[FINGERPRINT_VERSION] = rasa.__version__
+
+        # check the configuration
+
+        self._compare_raise_update(
+            fingerprint_key=FINGERPRINT_CONFIG_WITHOUT_EPOCHS_KEY,
+            new_fingerprint=self._get_fingerprint_of_config_without_epochs(config),
+            error_message=(
+                "Cannot finetune after making certain changes to the config. "
+                "The only changes that are allowed are"
+            ),
+        )
+
+        # check "core", i.e. domain
+
+        if core:
+            domain = importer.get_domain()
+            self._compare_raise_update(
+                fingerprint_key=FINGERPRINT_CORE,
+                new_fingerprint=self._get_fingerprint_of_domain_without_responses(
+                    domain
+                ),
+                error_message=(
+                    "Cannot finetune after making changes to the domain (only "
+                    "the responses may differ)."
+                ),
+            )
+
+        # check "nlu", i.e. nlu training data
+        # Note: Since this involves importing nlu data, it is probably the most
+        # costly step and hence happens last.
+
+        # NOTE: If there's a consistency check between domain and training data
+        # that ensures domain and NLU training data are consistent, then we can
+        # drop this check.
+
+        if nlu:
+            nlu_data = importer.get_nlu_data()
+            domain = importer.get_domain()
```

Comment:
Is this used?

Reply:
nope, copy&paste mistake - domain is not needed for nlu